### PR TITLE
Feat: system information collection logic in apiserver

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: wangyike <wangyike_wyk@163.com>

1. `get` userInfoCollection will return

```json
{
    "userID": "n9qpmg2wtkcbj4cc",
    "enableCollection": true
}
```
 
first `get` request will generate a UserID by random

2. post request `enable/disable`  will enable or disable user-Info-collection
3. write e2e-test

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/model/user_info_collection.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+*/
+
+package model
+
+func init() {
+	RegistModel(&UserInfoCollection{})
+}
+
+// UserInfoCollection userInfoCollection model
+type UserInfoCollection struct {
```
Comment (wonderflow) [inline pkg/apiserver/model/user_info_collection.go:24]:
```suggestion
type SystemInfoCollection struct {
```

File: pkg/apiserver/model/user_info_collection.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	BaseModel
+	UserID           string `json:"userID"`
```
Comment (wonderflow) [inline pkg/apiserver/model/user_info_collection.go:26]: InstallID

**Review by wonderflow — CHANGES_REQUESTED** [review]
rename overall

File: pkg/apiserver/model/user_info_collection.go (hunk 1 above)
Comment (barnettZQG) [reply pkg/apiserver/model/user_info_collection.go:24]: SystemInfo

Comment (wangyikewxgm) [reply pkg/apiserver/model/user_info_collection.go]: fixed

**Review by wonderflow — APPROVED** [review]

File: pkg/apiserver/rest/webservice/system_info.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	// Get
+	ws.Route(ws.DELETE("/").To(u.deleteSystemInfo).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Returns(200, "", apis.SystemInfoResponse{}).
+		Returns(400, "", bcode.Bcode{}).
+		Writes(apis.SystemInfoResponse{}))
+
+	// Get
+	ws.Route(ws.POST("/enable").To(u.enableSystemInfo).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/system_info.go:61]: The enable and disable API are not needed. Just provide an Update API for updating all system info.

**Comment by barnettZQG** [comment]
Don't have an API to get system version and statistics about app、cluster and addon?

File: pkg/apiserver/rest/webservice/system_info.go (hunk 3 above)
Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go:61]: 考虑过这种情况，但我觉的接口这种设计是合理的。因为第一次Get请求回创建一个installID有一种情况是在系统引导阶段用户需要直接disable掉collection，现在这种接口设计也是能够支持的。用过是update接口的话，你需要传一个installID，但这个时候还没有生成这个Id。

**Comment by wangyikewxgm** [comment]
> Don't have an API to get system version and statistics about app、cluster and addon?

We have discuss this problem offline, do  we. The dashboard can get these info from other apiserver interface. There is no need add other interface to return them.

File: pkg/apiserver/rest/webservice/system_info.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		Writes(apis.SystemInfoResponse{}))
+
+	// Get
+	ws.Route(ws.DELETE("/").To(u.deleteSystemInfo).
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Returns(200, "", apis.SystemInfoResponse{}).
+		Returns(400, "", bcode.Bcode{}).
+		Writes(apis.SystemInfoResponse{}))
+
+	// Get
```
[the remark this answers is not in the document]
Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go]: fixed

**Comment by barnettZQG** [comment]
> We have discuss this problem offline, do we?The dashboard can get these info from other apiserver interface. There is no need add other interface to return them.

There is no version info.

File: pkg/apiserver/rest/webservice/system_info.go (hunk 3 above)
Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/system_info.go:61]: SystemInfo 这个表以后是作为系统设置参数的存储，目前只有一个参数 “禁止数据收集”。

Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go:61]: version 是可以存在这个表当中，但这个值不应该由用户通过接口去修改才对吧

Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/system_info.go:61]: 版本不需要存储，直接读取，接口返回。

Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go:61]: > 版本不需要存储，直接读取，接口返回。

对，我也想到这个了，那就接口就保持这个样子吧， 只保留enable/disable 因为用户只能修改是否收集。但系统信息接口回返回version 数据。

Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/system_info.go:61]: 这个接口完全不符合寓意，建议直接变更为 PUT 接口哈。 SystemInfo 只会有一条数据，不需要用 installID 进行什么塞选哈。

Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go:61]: Put 请求body里传一整个systemInfo struct ，但只有enableCollection 字段会会生效？

Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/system_info.go:61]: {
  enableCollection: bool
}

Comment (wangyikewxgm) [reply pkg/apiserver/rest/webservice/system_info.go]: fixed

**Comment by wangyikewxgm** [comment]
> > We have discuss this problem offline, do we?The dashboard can get these info from other apiserver interface. There is no need add other interface to return them.
> 
> There is no version info.

Added version info.

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**